goshopify.go

```diff
@@ -424,3 +431,14 @@ func (c *Client) Put(path string, data, resource interface{}) error {
 func (c *Client) Delete(path string) error {
 	return c.CreateAndDo("DELETE", path, nil, nil, nil)
 }
+
+func GetAdminVersionedApiPathPrefix() string {
```

Comment:
Is there a reason we have to re-apply the validation on every call to this function? I think if we could validate the apiVersion a single time, then just return either admin, or the compiled version string that would be more performant.

Reply:
Cleanest path I see for this is to migrate that functionality directly into the NewClient function, and then access the data via a variable everywhere else for accessing the path prefix.